arangod/Pregel/Conductor/ClusterWorkerApi.cpp

```diff
@@ -1,22 +1,43 @@
 #include "ClusterWorkerApi.h"
+#include <variant>
 
 #include "Pregel/WorkerConductorMessages.h"
 #include "Pregel/NetworkConnection.h"
 
 using namespace arangodb::pregel::conductor;
 
+template<typename Out, typename In>
```

Comment:
A comment about why this is templated would not be too bad; The `In` parameter can be used to enforce what type of message to send, the `Out` parameter is used to say which type of response to expect.

(For context: my first thought was that one can just use `std::variant`throughout and be done with it, but in principle, statically asserting what type of message is expected as a response isn't a bad idea.